samplers.qmd

```diff
@@ -121,7 +121,216 @@ We can see from the autocorrelation function that we have to wait almost 100 ste
+
+With $weight_i$ being the size of the empirical VCV sample, we then weight the empirical VCV and initial VCV:
+$$\begin{align*}
+V_i = \frac{weight_i*sample\_vcv_i + \left(initial\_vcv\_weight+d+1\right)*initial\_vcv}{weight_i+initial\_vcv\_weight+d+2}
+\end{align*}$$
+(Note: weightings in numerator do not add up to denominator.)
+
+**2. Scaling**
+
+The shaping is controlled by the following input parameters to `monty_sampler_adaptive()`:
+
+- `initial_scaling`: the initial value for scaling ($\lambda_0$)
+- `min_scaling`: the minimum value for scaling
+- `scaling_increment`: the increment we use to increase or decrease the scaling
+- `acceptance_target`: the acceptance rate we target
+- `initial_scaling_weight`: value used in the starting denominator of the scaling update
+- `pre_diminish`: the number of iterations before we apply diminishing adaptation to the scaling updates
+- `log_scaling_change`: logical whether we update the scaling on a log scale or not
+- `adapt_end`: the final iteration at which we adaptively update the proposal VCV (also applies to shaping)
+
+
+`initial_scaling_weight` can be unspecified (`NULL`), in which case we use
+$$initial\_scaling\_weight = \frac{5}{acceptance\_target * (1 - acceptance\_target)}$$
+
+`scaling_increment` can be unspecified (`NULL`), in which case we use $scaling\_increment = \frac{1}{100}$ if $\log\_scaling\_change$ is $FALSE$, otherwise
+$$\begin{align*}
+scaling\_increment & = \left(1 - \frac{1}{d}\right) \left(\frac{\sqrt{2\pi}}{2A}\exp\left(\frac{A ^ 2}{2}\right)\right)\\ 
+& \quad \quad + \frac{1}{d * acceptance\_target * (1 - acceptance\_target)},
+\end{align*}$$
+where $A = -\psi^{-1}(acceptance\_target/2)$ and $\psi^{-1}$ is the inverse cdf of a standard normal distribution.
+
+
+We update scaling after iteration $i\leq adapt\_end$ based on the acceptance probability at that iteration $\alpha_i$ with 
+$$\begin{align*} 
+scaling\_change_i & = \frac{scaling\_increment}{\sqrt{scaling\_weight_i + \max\{0,\, i - pre\_diminish\}}}(\alpha_i - acceptance\_target)
+\end{align*}$$
+
+then
+
+- if $log\_scaling\_change = TRUE$:
+$$\begin{align*} 
+\log(\lambda_i) & = \max\left\{\log(min\_scaling), \log(\lambda_{i-1}) + scaling\_change_i\right\}
+\end{align*}$$
+- if $log\_scaling\_change = FALSE$:
+$$\begin{align*}
+\lambda_i = \max\left\{min\_scaling, \lambda_{i-1} + scaling\_change_i\right\}.
+\end{align*}$$
+
+So when the acceptance probability is larger than `acceptance_target` the scaling is increased, whereas when the acceptance probability is larger than `acceptance_target` the scaling is decreased. After $adapt\_end$ iterations, we no longer update the scaling.
+
+:::
+
+### Example: Adaptive vs. Non-adaptive Metropolis-Hastings
```

Comment:
I don't think it should be at the same level as the samplers - maybe #### ?

Reply:
Done